lib/widgets/expandable_fab.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/material.dart';
+
+class ExpandableFab extends StatefulWidget {
+  const ExpandableFab({Key? key, required this.items}) : super(key: key);
+
+  final List<ExpandableItem> items;
+
+  @override
+  _ExpandableFabState createState() => _ExpandableFabState();
+}
+
+class _ExpandableFabState extends State<ExpandableFab>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController _controller;
+  late final Animation<double> _animateIcon;
+  late final Animation<double> _expandAnimation;
+  late bool _open;
+
+  @override
+  void initState() {
+    _open = false;
+    _controller = AnimationController(
+      value: _open ? 1.0 : 0.0,
+      duration: const Duration(milliseconds: 250),
+      vsync: this,
+    );
+    _animateIcon = Tween<double>(begin: 0, end: 1).animate(_controller);
+    _expandAnimation = CurvedAnimation(
+      curve: Curves.easeIn,
+      reverseCurve: Curves.easeOut,
+      parent: _controller,
+    );
+    super.initState();
+  }
+
+  void _toggle() {
+    setState(() {
+      _open = !_open;
+      if (_open) {
+        _controller.forward();
+      } else {
+        _controller.reverse();
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.end,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: [
+        ...widget.items.map(
+          (e) => _ExpandingActionButton(
```

Comment:
chore: generally avoid single letter variables outside of exceptions especially with this since it is an `e` which is commonly used for a catch block.